pkg/image/prune/imagepruner.go

```diff
@@ -94,6 +96,9 @@ type PrunerOptions struct {
 	// KeepTagRevisions is the minimum number of tag revisions to preserve;
 	// revisions older than this value are candidates for pruning.
 	KeepTagRevisions int
+	// KeepOverSizeLimitImages indicates that images exceeding defined limits (openshift.io/Image)
+	// will not be considerate as candidates for pruning.
```

Comment:
In case of `keep-younger-then`, the image is excluded from pruning if younger than given age. Regardless of other flags. So the `keep` holds true always.

Prior to this patch, `keep-tag-revisions` preserved all the tags occurring prior to given index. It also excluded all the images referenced by preserved tags from pruning. So the `keep` held true always prior to this patch. With this patch, over sized images will be pruned regardless of this option. I don't like this patch impacting a semantic of the flag either. Can we deprecate it and choose a less consuming name?

As I said before, the `keep` in `keep-over-sized` doesn't give him any guarantees that the image will be kept.

I'm just afraid of bugs/issues, "why this image was deleted during prune if I wanted to **keep** it using this flag...". Replying him with "you should have do dry-run" isn't really comforting.